src/chronicle_storage.erl

```diff
@@ -158,25 +170,26 @@ open(PublishSnapshotFun) ->
 
     publish(Storage5).
 
-open_logs(#storage{data_dir = DataDir} = Storage) ->
-    {Orphans, Sealed, Current} = find_logs(DataDir),
-
-    NewStorage0 = Storage#storage{meta = #{},
-                                  pending_meta = #{},
-                                  config = undefined,
-                                  committed_config = undefined,
-                                  pending_configs = [],
-                                  low_seqno = ?NO_SEQNO + 1,
-                                  high_seqno = ?NO_SEQNO,
-                                  pending_high_seqno = ?NO_SEQNO,
-                                  current_snapshot = no_snapshot,
-                                  extra_snapshots = [],
-                                  log_segments = []},
-
-    %% TODO: Be more robust when dealing with corrupt log files. As long as a
-    %% consistent state can be recovered, corrupt logs should not be a fatal
-    %% error.
-    NewStorage1 =
+reset_storage(Storage) ->
+    ets:delete_all_objects(Storage#storage.log_tab),
+    ets:delete_all_objects(Storage#storage.log_info_tab),
+    Storage#storage{meta = #{},
+                    pending_meta = #{},
+                    config = undefined,
+                    committed_config = undefined,
+                    pending_configs = [],
+                    low_seqno = ?NO_SEQNO + 1,
+                    high_seqno = ?NO_SEQNO,
+                    pending_high_seqno = ?NO_SEQNO,
+                    current_snapshot = no_snapshot,
+                    extra_snapshots = [],
+                    log_segments = []}.
+
+open_logs(Sealed, Current, Storage) ->
```

Comment:
Since we don't use Storage in open_logs/3 (as it's immediately reset), we should not take it as an input parameter and just instantiate it and fill it in based on what we read in the logs and return it.

Reply:
We need publish_snapshot_fun and data_dir that are set up outside of open_logs

Reply:
got it